pkg/statser/channel_stats_watcher.go

```diff
@@ -0,0 +1,63 @@
+package statser
+
+import (
+	"context"
+	"time"
+
+	"github.com/atlassian/gostatsd"
+)
+
+// ChannelStatsWatcher reports metrics about channel usage to a Statser
+type ChannelStatsWatcher struct {
+	client   Statser
+	tags     gostatsd.Tags
+	capacity int
+	lenFunc  func() int
+	interval time.Duration
+}
+
+// NewChannelStatsWatcher creates a new ChannelStatsWatcher
+func NewChannelStatsWatcher(client Statser, channelName string, tags gostatsd.Tags, capacity int, lenFunc func() int, interval time.Duration) *ChannelStatsWatcher {
+	t := gostatsd.Tags{"channel:" + channelName}
+	if tags != nil {
+		t = append(t, tags...)
+	}
+	return &ChannelStatsWatcher{
+		client:   client,
+		tags:     t,
+		capacity: capacity,
+		lenFunc:  lenFunc,
+		interval: interval,
+	}
+}
+
+// Run will run a ChannelStatsWatcher in the background until the supplied context is
+// closed, or Stop is called.
+func (csw *ChannelStatsWatcher) Run(ctx context.Context) {
+	go func() {
```

Comment:
Let the caller decide whether they want it in a goroutine - this is much more flexible. Caller cannot undo `go` to make it run synchronously but can always add `go` to make it asynchronous. I think this is idiomatic in Go to make apis synchronous unless there are really good reasons not to.